element-desktop Wayland window doesn't show anymore

### Describe the bug
Since 796e82308363e1adc68d0982c02de8a705376c75 or a75daa8c05eec495fa48ba794f6f513916eece27 (not sure) the native Wayland window of element-desktop doesn't show anymore.

### Steps To Reproduce
When I launch element-desktop it shows a bunch of log messages but then just sits there without showing the window.

```console
$ rm -rf ~/.config/Element/
$ element-desktop --enable-features=UseOzonePlatform --ozone-platform=wayland
Keytar isn't installed; secure key storage is disabled.
Seshat isn't installed, event indexing is disabled.
/home/henri/.config/Element exists: no
/home/henri/.config/Riot exists: no
No update_base_url is defined: auto update is disabled
Fetching translation json for locale: en_EN
Resetting the UI components after locale change
Changing application language to en-us,en
Fetching translation json for locale: en-us
Fetching translation json for locale: en
Could not fetch translation json for locale: 'en' Error: Cannot find module './i18n/strings/en.json'
Require stack:
- /nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/language-helper.js
- /nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/tray.js
- /nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/electron-main.js
- /nix/store/qvnn6d6rb26z6n2bkl8m8gnlfcyp7bhz-electron-12.0.18/lib/electron/resources/default_app.asar/main.js
- 
    at Module._resolveFilename (internal/modules/cjs/loader.js:887:15)
    at Function.n._resolveFilename (electron/js2c/browser_init.js:261:1128)
    at Module._load (internal/modules/cjs/loader.js:732:27)
    at Function.f._load (electron/js2c/asar_bundle.js:5:12913)
    at Module.require (internal/modules/cjs/loader.js:959:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at AppLocalization.fetchTranslationJson (/nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/language-helper.js:76:20)
    at /nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/language-helper.js:89:39
    at Array.forEach (<anonymous>)
    at AppLocalization.setAppLocale (/nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/language-helper.js:88:17) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/language-helper.js',
    '/nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/tray.js',
    '/nix/store/nnv8sph0a16kzh2ds8qcxn4nji9v2x0w-element-desktop-1.8.2/share/element/electron/lib/electron-main.js',
    '/nix/store/qvnn6d6rb26z6n2bkl8m8gnlfcyp7bhz-electron-12.0.18/lib/electron/resources/default_app.asar/main.js',
    undefined
  ]
}
Resetting the UI components after locale change
```

The `Error: Cannot find module` is actually irrelevant here. That has been there for a long time, but I didn't notice any problems regarding i18n.

### Expected behavior
The window should show up.

### Screenshots
N/A

### Additional context
This could also be related to an Electron update.

### Notify maintainers
<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@ma27 @fadenb @mguentner @Ekleog @Ralith @mjlbach @dali99 @sumnerevans

### Metadata
Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

 - system: `"x86_64-linux"`
 - host os: `Linux 5.10.62, NixOS, 21.05.20210908.12eb1d1 (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.4pre20210802_47e96bb`
 - nixpkgs: `/etc/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: element-desktop
# a list of nixos modules affected by the problem
module:
```

--- a/pkgs/applications/networking/instant-messengers/element/element-desktop.nix
+++ b/pkgs/applications/networking/instant-messengers/element/element-desktop.nix
@@ -7,6 +7,8 @@
 , electron
 , element-web
 , callPackage
+, fetchpatch
+
 , Security
 , AppKit
 , CoreServices
@@ -31,6 +33,13 @@ in
 mkYarnPackage rec {
   name = "element-desktop-${version}";
   inherit version src;
+
+  patches = [
+    (fetchpatch {
+      url = "https://github.com/vector-im/element-desktop/commit/96e5389779f60c91b8fe80d7bd9af413d72ec61f.patch";
+      sha256 = "sha256-82I5BDNDWIfp+m2HpzTA5+39hMv2bTbmJlXfM4YUjDY=";
+    })
+  ];
 
   packageJSON = ./element-desktop-package.json;
   yarnNix = ./element-desktop-yarndeps.nix;

